ground/gcs/src/plugins/config/configoutputwidget.cpp

```diff
@@ -455,15 +465,78 @@ void ConfigOutputWidget::refreshWidgetsValues(UAVObject * obj)
     {
         quint32 minValue = actuatorSettingsData.ChannelMin[outputChannelForm->index()];
         quint32 maxValue = actuatorSettingsData.ChannelMax[outputChannelForm->index()];
+
         outputChannelForm->setMinmax(minValue, maxValue);
 
         quint32 neutral = actuatorSettingsData.ChannelNeutral[outputChannelForm->index()];
         outputChannelForm->setNeutral(neutral);
     }
+
+    refreshWidgetRanges();
+}
+
+void ConfigOutputWidget::refreshWidgetRanges()
+{
+    QList<OutputChannelForm*> outputChannelForms = findChildren<OutputChannelForm*>();
+
+    ExtensionSystem::PluginManager *pm = ExtensionSystem::PluginManager::instance();
+    Q_ASSERT(pm);
+    UAVObjectUtilManager* utilMngr = pm->getObject<UAVObjectUtilManager>();
+
+    if (utilMngr) {
+        Core::IBoardType *board = utilMngr->getBoardType();
+        if (board != NULL) {
+            QVector< QVector<qint32> > banks = board->getChannelBanks();
+
+            for (int i=0; i < banks.size() ; i++) {
+                QVector<int> bank = banks[i];
+
+                // Pull out the parameters relating to this bank, and
+                // calculate the maximum value.
+                //
+                QComboBox* ccmb = rateList.at(i);
+                quint32 timerFreq = timerStringToFreq(ccmb->currentText());
+
+                QComboBox* res = resList.at(i);
+                quint32 timerRes = res->currentIndex();
+
+                double maxPulseWidth = 0, uavoMaxPulseWidth = 0;
+
+                // Saturate at the UAVO's maximum value
```

Comment:
This comment/variable is inaccurate I think. The UAVO max is in microseconds. However, you are actually calculating the maximum number of microseconds on the physical timer, assuming 16 bits (some aren't, but a reasonable conservative assumption).

So it is really timerMaxPeriodUs